### DISCUSSION TITLE
STM32F411 ADC sequence capture hangs when acquisition time is set to ADC_ACQ_TIME_DEFAULT

### DISCUSSION THREAD

**Discussion by fcgdam** [zephyrproject-rtos/zephyr#79253] [Q&A]
Just asking if anyone can clarify what might be the problem.

I'm testing an ADC capture with the STM32F411 Weact blackpill board, using the zephyr example adc-sequence.

If on the overlay file to define the ADC channels, only one channel is defined, and the property **zephyr,acquisition-time** is set to the **ADC_ACQ_TIME_DEFAULT**, it works fine.

But as soon as I add another channel, to the DTS overlay:

```
&adc1 {
    #address-cells = <1>;
    #size-cells = <0>;
    channel@0 {
        reg = <0>;
        zephyr,gain = "ADC_GAIN_1";
        zephyr,reference = "ADC_REF_INTERNAL";
        /*zephyr,acquisition-time = <ADC_ACQ_TIME_MAX>;*/
        zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
        zephyr,resolution = <12>;
    };

    channel@1 {
        reg = <1>;
        zephyr,gain = "ADC_GAIN_1";
        zephyr,reference = "ADC_REF_INTERNAL";
        zephyr,acquisition-time = <ADC_ACQ_TIME_DEFAULT>;
        zephyr,resolution = <12>;
    };
};
```

The **err = adc_read(adc, &sequence);** hangs/block indefinitely.

IF however I change the acquisition-time to **ADC_ACQ_TIME_MAX**, it starts to work just fine.

 Any ideas why this behavior is as such?

**Comment by gautierg-st** [comment]
Hello @fcgdam,
You probably have an overrun error. The sampling rate is too fast comparing to the speed the software can read them.
Increasing the sampling time lets more time to the software to read the first before the second arrive.
You can also play with the ADC prescaler to decrease the ADC source clock frequency (on F411 you can choose 2/4/6/8) to avoid triggering into an overrun.

**Comment by fcgdam** [comment]
Hi, don't know if it was or is an overflow error, but indeed, changing the device tree ADC definition value from the default value of 2 for the prescaler to **st,adc-prescaler = <4>;** and testing the possible acquisition time values does work now.

For example, if the prescaler is set to 2, the default value, and the acquisition time is set to 112 (**zephyr,acquisition-time = <	ADC_ACQ_TIME(ADC_ACQ_TIME_TICKS, 112) >;** it hangs.

But changing the prescaler to 4, it starts to work.

So it is indeed a timing issue.
